### PR TITLE
Refactor move nav-native -> SDK type mapping and roadobject models to core module

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Fixes https://github.com/mapbox/mapbox-navigation-android/issues/3735

Follow up from https://github.com/mapbox/mapbox-navigation-android/pull/4113

Refactors moves nav-native -> SDK type mapping and `roadobject` models to core module

### Changelog

```
<changelog>Refactored moved nav-native -> SDK type mapping and `roadobject` models to core module.</changelog>
```

### CODE REVIEW

**Review by korshaknn — APPROVED** [review]

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt [hunk 1]
```diff
@@ -10,15 +10,15 @@ object RestStopType {
      * This means that the type is either erroneous,
      * or there's a new type available server-side which will be exposed in the SDK in the future.
      */
-    const val Unknown = -1
+    const val UNKNOWN = -1
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:13]: could you also add `@StringDef` as a part of the PR

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/tollcollection/TollCollectionType.kt [hunk 2]
```diff
@@ -10,15 +10,15 @@ object TollCollectionType {
...
      * Describes a payment booth.
      */
-    const val TollBooth = 1
+    const val TOLL_BOOTH = 1
 
     /**
      * Describes an overhead electronic gantry.
      */
-    const val TollGantry = 2
+    const val TOLL_GANTRY = 2
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/tollcollection/TollCollectionType.kt:23]: `@IntDef` candidate

**Review by RingerJK — APPROVED** [review]
lgtm, would be great to group consts by `Def`s

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt [hunk 3]
```diff
@@ -10,15 +12,26 @@ object RestStopType {
...
     /**
      * Describes a service area.
      */
-    const val ServiceArea = 2
+    const val SERVICE_AREA = 2
+
+    /**
+     * Rest stop type.
+     */
+    @Retention(AnnotationRetention.SOURCE)
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:30]:
```suggestion
    @Retention(AnnotationRetention.RUNTIME)
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/tollcollection/TollCollectionType.kt [hunk 4]
```diff
@@ -10,15 +12,26 @@ object TollCollectionType {
...
+
+    /**
+     * Toll collection type.
+     */
+    @Retention(AnnotationRetention.SOURCE)
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/tollcollection/TollCollectionType.kt:30]:
```suggestion
    @Retention(AnnotationRetention.RUNTIME)
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt (hunk 3 above)
Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:30]: We use `@Retention(AnnotationRetention.SOURCE)` across our codebase. Should we change all of them? 🤔

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:30]: @RingerJK  Going to go ahead and merge here. We can discuss further, could you cut a ticket when you have a chance? 🙏

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:30]: > We use `@Retention(AnnotationRetention.SOURCE)` across our codebase. Should we change all of them? 🤔

yeah, definitely for public API

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/roadobject/reststop/RestStopType.kt:30]: following up at https://github.com/mapbox/mapbox-navigation-android/issues/4145

**Pull request merged**